src/python/pants/backend/openapi/util_rules/generator_process.py

```diff
@@ -23,15 +22,10 @@
 from pants.util.logging import LogLevel
 
 
-@unique
-class OpenAPIGeneratorType(Enum):
-    JAVA = "java"
-
-
```

Comment:
If following the previously suggested approach of having a validation rule for the generator name. This could be replaced by a mapping that gets cached:

```python
@dataclass(frozen=True)
class OpenAPIGeneratorNameMapping:
    mapping: FrozenDict[str, tuple[str, ...]]

    def generators_for(self, lang: str) -> tuple[str, ...]:
          valid_choices = self.mapping.get(lang)
          if valid_choices is None:
              raise ValueError(f"Unsupported language generator: {lang}")
          return valid_choices

    def all_languages(self) -> tuple[str, ...]:
       # return a tuple with the supported languages

    def all_generator(self) -> tuple[str, ...]:
       # return a tuple with all the generator names.

async get_openapi_generator_name_mapping(openapi_generator: OpenAPIGenerator) -> OpenAPIGeneratorNameMapping:
    # Run the JVM process for the generator passing arguments `list -s`
    # Parse CSL result from stdout. Generator names are always prefixed by the language name, e.i.: `java`, `java-helicon-client`, `python`, etc.
    # https://openapi-generator.tech/docs/generators
```

Then the `OpenAPIGeneratorNameMapping` could be used as the basis for a validation of the `generator_name` field at different stages.

Reply:
Added list of names `OpenAPIGeneratorNames`, mapping to languages might create more problems